serverless/networking/serverless-ossm-tls.adoc

```diff
@@ -0,0 +1,182 @@
+:context: serverless-ossm-tls
+include::modules/common-attributes.adoc[]
+
+toc::[]
+
+The following example shows how you can create a TLS key and certificates for a custom domain and subdomain.
+
+.Prerequisites
+
+* Install xref:../../serverless/installing_serverless/installing-openshift-serverless.adoc#installing-openshift-serverless[{ServerlessProductName}].
+* Install xref:../../service_mesh/v1x/installing-ossm.adoc#installing-ossm-v1x[{ProductName}].
+* Configure a custom domain. See xref:../../serverless/networking/serverless-ossm-custom-domains.adoc#serverless-ossm-custom-domains[Using custom domains for Knative services with Service Mesh].
+* In this example, `openssl` is used to generate certificates, but you can use any certificate generation tool to create these.
+
+[IMPORTANT]
+====
+This example uses the `example.com` domain.
+The example certificate for this domain is used as a certificate authority (CA) that signs the sub-domain certificate.
+
+To complete and verify this procedure in your deployment, you will need either a certificate signed by a widely trusted public CA, or a CA provided by your organization.
+
+Example commands should be adjusted according to your domain, sub-domain and CA.
+====
+
+.Procedure
+
+. Create a root certificate and private key to sign the certificates for your services:
++
+[source,terminal]
+----
+$ openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 \
+    -subj '/O=Example Inc./CN=example.com' \
+    -keyout example.com.key \
+    -out example.com.crt
+----
+. Create a certificate signing request for your domain:
++
+[source,terminal]
+----
+$ openssl req -out custom.example.com.csr -newkey rsa:2048 -nodes \
+    -keyout custom.example.com.key \
+    -subj "/CN=custom-ksvc-domain.example.com/O=Example Inc."
+----
+. Sign the request with your CA:
++
+[source,terminal]
+----
+$ openssl x509 -req -days 365 -set_serial 0 \
+    -CA example.com.crt \
+    -CAkey example.com.key \
+    -in custom.example.com.csr \
+    -out custom.example.com.crt
+----
+. Check that the certificates appear in your directory:
++
+[source,terminal]
+----
+$ ls -1
+----
++
+.Example output
+[source,terminal]
+----
+custom.example.com.crt
+custom.example.com.csr
+custom.example.com.key
+example.com.crt
+example.com.key
+----
+. Create a secret:
++
+[source,terminal]
+----
+$ oc create -n istio-system secret tls custom.example.com \
+    --key=custom.example.com.key \
+    --cert=custom.example.com.crt
+----
+. Attach the secret to the Istio ingress gateway, by editing the `ServiceMeshControlPlane` resource.
+.. Edit the `ServiceMeshControlPlane` resource:
++
+[source,terminal]
+----
+$ oc edit -n istio-system ServiceMeshControlPlane <control_plane_name>
+----
+.. Check that the following lines exist in the resource, and if they do not, add them:
++
+[source,yaml]
+----
+istio-ingressgateway:
+  secretVolumes:
+  - mountPath: /custom.example.com
+    name: custom-example-com
+    secretName: custom.example.com
```

Comment:
This configuration is ignored on SM 1.x. 

We need the following configuration for SM 2.x:

```
spec:
  version: v2.0
  gateways:
    ingress:
      volumes:
      - volume:
          secret:
            secretName: custom.example.com
        volumeMount:
          name: custom-example-com
          mountPath: /custom.example.com
```

Reply:
I got you. Does this mean it doesn't work on v1 at all?

We might need two different service mesh sections if that's the case, but that's OK, and it might not be a bad thing timing wise. There's an upstream docs issue re splitting up the information about networking layers, and checking that we're covering the ones that can be used with Knative, so maybe we can do both together?

Would you be able and willing to provide input @nak3 ? Is there anyone else who is a good SME for this just so we're not overloading you? :slightly_smiling_face: I'm happy to doc as much info as you can give me, and re the comment below, let's have two versions if we need them or else just update everything for 2.x so we're consistent. Little more time consuming, but worth it to get it right IMHO.

Upstream related issue: https://github.com/knative/docs/issues/3117

Reply:
>  Does this mean it doesn't work on v1 at all?

Yes, you correct. It doesn't work on v1 at all.